### ISSUE TITLE
Multiple VeSync API calls

### ISSUE THREAD

**Issue by borpin** [vlebourl/custom_vesync#3]
I have been getting these errors. I've raised an issue on the VeSync library, but from these errors, it looks like the integration is calling the API for every entity which would not be good if the API is rate limited. I note there was some discussion about this on the Core PR page.

I wonder if something different needs to be done so different entities get updated by the same API call.

This also assumes on my part that it is the same API call for all read entities.

```
2022-04-02 12:16:22 ERROR (MainThread) [homeassistant.helpers.entity] Update for sensor.study_hum_current_humidity fails
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 535, in async_update_ha_state
    await self.async_device_update()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 744, in async_device_update
    raise exc
  File "/usr/local/lib/python3.9/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/config/custom_components/vesync2/common.py", line 124, in update
    self.device.update()
  File "/usr/local/lib/python3.9/site-packages/pyvesync/vesyncfan.py", line 965, in update
    self.get_details()
  File "/usr/local/lib/python3.9/site-packages/pyvesync/vesyncfan.py", line 946, in get_details
    outer_result = r.get('result', {})
AttributeError: 'NoneType' object has no attribute 'get'
2022-04-02 12:16:22 ERROR (MainThread) [homeassistant.helpers.entity] Update for binary_sensor.study_hum_out_of_water fails
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 535, in async_update_ha_state
    await self.async_device_update()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 744, in async_device_update
    raise exc
  File "/usr/local/lib/python3.9/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/config/custom_components/vesync2/common.py", line 124, in update
    self.device.update()
  File "/usr/local/lib/python3.9/site-packages/pyvesync/vesyncfan.py", line 965, in update
    self.get_details()
  File "/usr/local/lib/python3.9/site-packages/pyvesync/vesyncfan.py", line 946, in get_details
    outer_result = r.get('result', {})
AttributeError: 'NoneType' object has no attribute 'get'
2022-04-02 12:16:22 ERROR (MainThread) [homeassistant.helpers.entity] Update for number.study_hum_mist_level fails
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 535, in async_update_ha_state
    await self.async_device_update()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 744, in async_device_update
    raise exc
  File "/usr/local/lib/python3.9/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/config/custom_components/vesync2/common.py", line 124, in update
    self.device.update()
  File "/usr/local/lib/python3.9/site-packages/pyvesync/vesyncfan.py", line 965, in update
    self.get_details()
  File "/usr/local/lib/python3.9/site-packages/pyvesync/vesyncfan.py", line 946, in get_details
    outer_result = r.get('result', {})
AttributeError: 'NoneType' object has no attribute 'get'
2022-04-02 12:16:23 ERROR (MainThread) [homeassistant.helpers.entity] Update for switch.study_hum_display fails
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 535, in async_update_ha_state
    await self.async_device_update()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 744, in async_device_update
    raise exc
  File "/usr/local/lib/python3.9/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/config/custom_components/vesync2/common.py", line 124, in update
    self.device.update()
  File "/usr/local/lib/python3.9/site-packages/pyvesync/vesyncfan.py", line 965, in update
    self.get_details()
  File "/usr/local/lib/python3.9/site-packages/pyvesync/vesyncfan.py", line 946, in get_details
    outer_result = r.get('result', {})
AttributeError: 'NoneType' object has no attribute 'get'
2022-04-02 12:16:27 ERROR (MainThread) [homeassistant.helpers.entity] Update for binary_sensor.study_hum_water_tank_lifted fails
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 535, in async_update_ha_state
    await self.async_device_update()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 744, in async_device_update
    raise exc
  File "/usr/local/lib/python3.9/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/config/custom_components/vesync2/common.py", line 124, in update
    self.device.update()
  File "/usr/local/lib/python3.9/site-packages/pyvesync/vesyncfan.py", line 965, in update
    self.get_details()
  File "/usr/local/lib/python3.9/site-packages/pyvesync/vesyncfan.py", line 946, in get_details
    outer_result = r.get('result', {})
AttributeError: 'NoneType' object has no attribute 'get'
2022-04-02 12:16:28 ERROR (MainThread) [homeassistant.helpers.entity] Update for number.study_hum_target_level fails
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 535, in async_update_ha_state
    await self.async_device_update()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 744, in async_device_update
    raise exc
  File "/usr/local/lib/python3.9/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/config/custom_components/vesync2/common.py", line 124, in update
    self.device.update()
  File "/usr/local/lib/python3.9/site-packages/pyvesync/vesyncfan.py", line 965, in update
    self.get_details()
  File "/usr/local/lib/python3.9/site-packages/pyvesync/vesyncfan.py", line 946, in get_details
    outer_result = r.get('result', {})
AttributeError: 'NoneType' object has no attribute 'get'
2022-04-02 12:16:28 ERROR (MainThread) [homeassistant.helpers.entity] Update for switch.study_hum_automatic_stop fails
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 535, in async_update_ha_state
    await self.async_device_update()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 744, in async_device_update
    raise exc
  File "/usr/local/lib/python3.9/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/config/custom_components/vesync2/common.py", line 124, in update
    self.device.update()
  File "/usr/local/lib/python3.9/site-packages/pyvesync/vesyncfan.py", line 965, in update
    self.get_details()
  File "/usr/local/lib/python3.9/site-packages/pyvesync/vesyncfan.py", line 946, in get_details
    outer_result = r.get('result', {})
AttributeError: 'NoneType' object has no attribute 'get'
2022-04-02 12:16:33 ERROR (MainThread) [homeassistant.helpers.entity] Update for switch.study_hum_auto_mode fails
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 535, in async_update_ha_state
    await self.async_device_update()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 744, in async_device_update
    raise exc
  File "/usr/local/lib/python3.9/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/config/custom_components/vesync2/common.py", line 124, in update
    self.device.update()
  File "/usr/local/lib/python3.9/site-packages/pyvesync/vesyncfan.py", line 965, in update
    self.get_details()
  File "/usr/local/lib/python3.9/site-packages/pyvesync/vesyncfan.py", line 946, in get_details
    outer_result = r.get('result', {})
AttributeError: 'NoneType' object has no attribute 'get'
```

**Comment by vlebourl** [comment]
```py
AttributeError: 'NoneType' object has no attribute 'get'
```
was resolved by #12 

I'll close this issue. So far I haven't seen anything regarding API limitaitons. We'll open a new one if that rises.

**Comment by borpin** [comment]
Ok, it was this comment and the subsequent discussion.

I can see that

- The error will have been fixed by the change to the library

- Subsequent discussion suggested it might be a different problem.

One thing, if there is no response from the server, is an error propagated to the logs?

If not there probably should be.
